app/assets/javascripts/modules/repositories/services/vulnerabilities-parser.js

```diff
@@ -1,32 +1,23 @@
-const AVAILABLE_BACKENDS = ['clair', 'zypper', 'dummy'];
-
-const parse = function (vulnerabilities) {
+const countBySeverities = function (vulnerabilities) {
   const severities = {
-    High: 0,
-    Normal: 0,
+    Negligible: 0,
+    Unknown: 0,
     Low: 0,
+    Medium: 0,
+    High: 0,
+    Critical: 0,
+    Defcon1: 0,
   };
-  let total = 0;
 
   if (vulnerabilities) {
-    AVAILABLE_BACKENDS.forEach((backend) => {
-      if (!vulnerabilities[backend]) {
-        return;
-      }
-
-      vulnerabilities[backend].forEach((vul) => {
-        severities[vul.severity] += 1;
-        total += 1;
-      });
+    vulnerabilities.forEach((vul) => {
+      severities[vul.severity] += 1;
```

Comment:
Overstretching a bit, but what if `vul.severity` returns something which is not covered by the `severities` hash (because of an update on clair or something like that)?. I think we should have a default case (maybe using `Unknown` ?).

Reply:
Well, these severities are based on [CVSS rating scale](https://www.first.org/cvss/specification-document#5-Qualitative-Severity-Rating-Scale). So I don't think this will be changed that often.